codec/src/main/java/io/netty/handler/codec/ByteStringValueConverter.java

```diff
@@ -0,0 +1,134 @@
+    private static final Charset DEFAULT_CHARSET = CharsetUtil.UTF_8;
+
+    @Override
+    public ByteString convertObject(Object value) {
+        if (value instanceof ByteString) {
+            return (ByteString) value;
+        }
+        if (value instanceof CharSequence) {
+            return new ByteString((CharSequence) value, DEFAULT_CHARSET);
+        }
+        return new ByteString(value.toString(), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertInt(int value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertLong(long value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertDouble(double value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertChar(char value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertBoolean(boolean value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public ByteString convertFloat(float value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public int convertToInt(ByteString value) {
+        return value.parseAsciiInt();
+    }
+
+    @Override
+    public long convertToLong(ByteString value) {
+        return value.parseAsciiLong();
+    }
+
+    @Override
+    public ByteString convertTimeMillis(long value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public long convertToTimeMillis(ByteString value) {
+        try {
+            return HeaderDateFormat.get().parse(value.toString());
+        } catch (ParseException e) {
+            PlatformDependent.throwException(e);
+        }
+        return 0;
+    }
+
+    @Override
+    public double convertToDouble(ByteString value) {
+        return value.parseAsciiDouble();
+    }
+
+    @Override
+    public char convertToChar(ByteString value) {
+        return value.parseChar();
+    }
+
+    @Override
+    public boolean convertToBoolean(ByteString value) {
+        return value.byteAt(0) != 0;
+    }
+
+    @Override
+    public float convertToFloat(ByteString value) {
+        return value.parseAsciiFloat();
+    }
+
+    @Override
+    public ByteString convertShort(short value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public short convertToShort(ByteString value) {
+        return value.parseAsciiShort();
+    }
+
+    @Override
+    public ByteString convertByte(byte value) {
+        return new ByteString(String.valueOf(value), DEFAULT_CHARSET);
+    }
+
+    @Override
+    public byte convertToByte(ByteString value) {
```

Comment:
IIUC what @normanmaurer says is to throw an exception with more contextually meaningful message. i.e. IOOBE vs IAE('value' is empty)